### PR TITLE
Snap to stage

### PR DESCRIPTION
**Pull request by Unclesdad**
Have you ever found it tedious to find the exact angle for which climbing would be most efficient? Well worry not, because snapToStage() is now here! With one press of the A button on the driver's controller, it'll snap to the nearest chain, looking directly at it! Here's a review from one of our users:
Warren: "wow thats awesome thats pretty cool"
Ivan: "you did it" (clap clap clap clap clap)
Get yours today at only one pull! That's right, just one pull into main!

### CODE REVIEW

File: src/main/java/org/sciborgs1155/robot/Constants.java [hunk 1]
```diff
@@ -92,13 +94,34 @@ public static class Field {
...
+        new Translation2d(Inches.of(472.4375), Inches.of(133.95));
+
     /** Returns the translation of the speaker for the robot's alliance. */
     public static Translation3d speaker() {
       return alliance() == Alliance.Red
           ? RED_SPEAKER_POSE.plus(TARGET_OFFSET.rotateBy(new Rotation3d(0, 0, Math.PI)))
           : BLUE_SPEAKER_POSE.plus(TARGET_OFFSET);
     }
 
+    public static List<Translation2d> chainCoordinateList() {
```
Comment (anglesideangle) [inline src/main/java/org/sciborgs1155/robot/Constants.java]: call this `chainCoordinates` and add a line of javadoc for consistency

File: src/main/java/org/sciborgs1155/robot/drive/SparkModule.java [hunk 2]
```diff
@@ -82,6 +82,9 @@ public SparkModule(int drivePort, int turnPort, Rotation2d angularOffset) {
     resetEncoders();
 
     this.angularOffset = angularOffset;
+
+    lastVelocity = driveEncoder.getVelocity();
```
Comment (anglesideangle) [inline src/main/java/org/sciborgs1155/robot/drive/SparkModule.java]: not sure why these changes are here

File: src/main/java/org/sciborgs1155/robot/commands/Climbing.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   *     units.
+   */
+  public Rotation2d findChainAngle(/* Translation2d coordinates */ ) {
+    Translation2d nearest;
+    Translation2d coords = drive.pose().getTranslation();
+
+    nearest = coords.nearest(chainCoordinateList());
+
+    // TODO forgive me lord for I have sinned
+    if (nearest == BLUE_STAGE_AMPSIDE) {
```
Comment (anglesideangle) [inline src/main/java/org/sciborgs1155/robot/commands/Climbing.java]: you can cut this in half by only dealing with the current alliance's stage

Comment (anglesideangle) [reply src/main/java/org/sciborgs1155/robot/commands/Climbing.java]: actually, you can make the chain coordinates `Pose2d`s with the proper rotation inside them, and return the rotation from the nearest pose without the if statement

File: src/main/java/org/sciborgs1155/robot/commands/Climbing.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      return Rotation2d.fromRadians(Math.PI * 2 / 3);
+    } else if (nearest == RED_STAGE_MIDSIDE) {
+      return Rotation2d.fromRadians(0);
+    }
+    return Rotation2d.fromRadians(0);
+  }
+
+  /** Turns the robot such that its heading is perpendicular to its nearest stage climbing chain. */
+  public Command snapToStage(
+      DoubleSupplier vx, DoubleSupplier vy /*, Supplier<Translation2d> coords*/) {
```
Comment (anglesideangle) [inline src/main/java/org/sciborgs1155/robot/commands/Climbing.java]: remove the unnecessary commented out arguments

**Review by anglesideangle — CHANGES_REQUESTED** [review]

File: src/main/java/org/sciborgs1155/robot/drive/SparkModule.java (hunk 2 above)
Comment (Unclesdad) [reply src/main/java/org/sciborgs1155/robot/drive/SparkModule.java]: merge conflict issue

File: src/main/java/org/sciborgs1155/robot/commands/Climbing.java (hunk 3 above)
Comment (Unclesdad) [reply src/main/java/org/sciborgs1155/robot/commands/Climbing.java]: oh my god the entire time it coudlve been a single line i want to end it all

File: src/main/java/org/sciborgs1155/robot/commands/Climbing.java (hunk 4 above)
Comment (Unclesdad) [reply src/main/java/org/sciborgs1155/robot/commands/Climbing.java]: aight

**Review by anglesideangle — APPROVED** [review]
Wow! This pull request is revolutionary. It is truly a momentous improvement to climbing
